optimade/filtertransformers/mongo.py

```diff
@@ -389,6 +393,43 @@ def replace_with_relationship(subdict, prop, expr):
             filter_, check_for_entry_type, replace_with_relationship
         )
 
+    def _apply_unknown_or_null_filter(self, filter_: dict) -> dict:
+        """ This method loops through the query and replaces the check for
+        KNOWN with a check for existence and a check for not null, and the
+        inverse for UNKNOWN.
+
+        """
+
+        def check_for_known_filter(prop, expr):
+            """ Find cases where the query dict looks like
+            `{"field": {"#known": T/F}}` or
+            `{"field": "$not": {"#known": T/F}}`, which is a magic word
+            for KNOWN/UNKNOWN filters in this transformer.
+
+            """
+            return isinstance(expr, dict) and (
+                "#known" in expr or "#known" in expr.get("$not", {})
```

Comment:
It only executes this part if `isinstance(expr, dict)` is `True`, right? I think it does.

Reply:
Yup, `if a() and b()` only evaluates `b()` if `a()` is `True`.